### PR TITLE
Add and Remove Books

### PR DESCRIPTION
**Pull request by rpire**
## Changes made:
- Configure `Redux` store, constants, and reducers.
- Created `addBook` and `removeBook` action creators.
- Refactor components to work with the `Redux` logic.

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
 const Book = (props) => {
-  const { book, handleChange } = props;
-  const { author, title, category } = book;
+  const { book } = props;
+  const {
+    id,
+    author,
+    title,
+    category,
+  } = book;
```
Comment (EricMbouwe) [inline src/components/Book.js]: - [x] Here you are repeating yourself by doing props destructuring twice. Let's be 'DRY' and have it in a single statement.
  `const { id, author, title, category } = props;`

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @rpire, 🖐

Good job so far, you've done pretty much all the requirements for this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rpire** [comment]
Hello, @EricMbouwe 👋

Thank you for reviewing my work. You are right, the code is not completely "DRY". 😯

I will make sure to implement the requested changes before submitting my project for another review.

Cheers and Happy Coding! 🤓

File: src/components/Book.js (hunk 1 above)
Comment (rpire) [reply src/components/Book.js]: Requested change fulfilled.

> Note that the props are being passed within `book` key. So the correct destructuring assignment is:
> `const { book: { id, author, title, category } } = props`

**Review by dasileker — APPROVED** [review]
### **Hi @rpire,** 

### **Project Approved ✅**


Approved


Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rpire** [comment]
Hi, @dasileker 👋

### Thank you! 🙌

Thank you for reviewing my work. I will proceed to merge the changes into the `dev` branch. 🚀

Cheers and Happy Coding! 🤓

**Pull request merged**